src/Silex/Application.php

```diff
@@ -64,8 +63,12 @@ public function __construct(array $values = array())
 
         $app = $this;
 
-        $this['routes'] = function () {
-            return new RouteCollection();
+        $this['routes_class'] = 'Symfony\\Component\\Routing\\RouteCollection';
+        $this['routes_factory'] = $this->factory(function () use ($app) {
```

Comment:
a factory is needed if ControllerCollection reuses it.

However @lschricke, this PR won't solve things for you if a ControllerCollection defines the same route name than another one (or that the main route collection)